### ISSUE TITLE
Crash during initial storage location setup prevents successfully completing initial setup

### ISSUE THREAD

**Issue by alfieyoko2** [aniyomiorg/aniyomi#1386]
### Steps to reproduce

|1. Initial screen.|2. Second screen with option to select the storage location |3. Android file picker opens|
|-|-|-|
|![Screenshot_20240114_022629_Aniyomi](https://github.com/aniyomiorg/aniyomi/assets/158510175/e28b934a-7339-4039-9b4a-e1aa3c13cfda)|![Step2](https://github.com/aniyomiorg/aniyomi/assets/158510175/d2b0c940-0f0f-4f18-8c28-d808f29cf29d) |![Screenshot_20240114_022658_Files](https://github.com/aniyomiorg/aniyomi/assets/158510175/34cc691b-774a-4fdf-8338-174f2156020a) |
|4. Opening the folder ("Mihon", "Aniyomi" or "Tachiyomi" depending on the app)|6. After pressing the select button, it asks for read/write permission for the selected folder. On clicking allow, the app crashes|7. The initial screen where the app ends up after the crash. All progress lost and no folder selected, so it doesn't let you continue to the main part of the app|
|![Screenshot_20240114_022710_Files](https://github.com/aniyomiorg/aniyomi/assets/158510175/9584a632-38ac-4c86-92cd-b0a93cb9226a)|![Screenshot_20240114_022719_Files](https://github.com/aniyomiorg/aniyomi/assets/158510175/dc2d136d-6595-40de-8d17-18e7c2dd4847)|![Screenshot](https://github.com/aniyomiorg/aniyomi/assets/158510175/eee24c66-8672-4ef9-8a68-2da26b18dde3)|

These screenshots are from testing it on Aniyomi, but the bug is 100% identically replicable with at least Mihon 0.16.3, Tachiyomi from 0.15.x and Aniyomi from 0.15.x

Maybe a compatibility issue with One UI?

Maybe there's a possibility of a temporary fix that lets you skip this initial setup? (and maybe an option where it can be done later in settings still, which would be useful anyway so the storage folder can be changed after setup)

Bug verified on a Samsung Galaxy S22 Ultra, Security Patch level 1.1.2024

### Expected behavior

After Step 5, it should return to the screen of Step 2, with the selected storage location locked in and the option to continue appearing.

### Actual behavior

App seems to crash at Step 5, and then immediately reverting back to Step 1 (same as Step 6), not having locked in the selected storage location, so there is no way to successfully finish the initial setup.

### Aniyomi version

0.15.2.4

### Android version

Android 14, Security Patch Level 1.1.2024 (One UI 6.0)

### Device

Samsung Galaxy S22 Ultra

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an official extension, I should be opening an issue in the [extensions repository](https://github.com/aniyomiorg/aniyomi-extensions/issues/new/choose).
- [X] If this is an issue with an official manga extension and this issue can be replicated in the Tachiyomi app, that I should be opening an issue in [Tachiyomi's extensions repository](https://github.com/tachiyomiorg/extensions/issues/new/choose).
- [X] I have gone through the [FAQ](https://aniyomi.org/docs/faq/general) and [troubleshooting guide](https://aniyomi.org/docs/guides/troubleshooting/).
- [X] I have updated the app to version **[0.15.2.4](https://github.com/aniyomiorg/aniyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

**Comment by alfieyoko2** [comment]
UPDATE:
Issue is solved, the crash happened because "Don't keep activities" was enabled in Developer Settings.
Disabling that setting makes it work normally.
